c++/src/kj/common.h

```diff
@@ -373,6 +373,23 @@ struct DisallowConstCopy {
   DisallowConstCopy& operator=(DisallowConstCopy&&) = default;
 };
 
+#if _MSC_VER
+
+#define MSVC_CAPTURE(obj) obj=::kj::cp(obj)
```

Comment:
I would call this `KJ_CPCAP()`:
* KJ macros should start with `KJ_` to follow macro namespacing rules.
* There is actually a macro called `KJ_MVCAP()` in the Sandstorm codebase which has a similar purpose but for capture-by-move.

More generally, the need for this macro is unfortunate. I've actually been thinking of giving up on the non-const-copy thing. The purpose was supposed to be to enforce "transitive const" semantics, but in practice it causes quite a lot of problems, including tickling a lot of compiler bugs (there's actually an ABI incompatibility between Clang and GCC triggered by it). Perhaps transitive-const simply is not practical to enforce in C++. Perhaps it's time to do that.

I guess this commit should stay in for now but I'll revert it when I'm ready to roll back transitive-const enforcement.